src/app/app.component.ts

```diff
@@ -193,21 +202,18 @@ export class AppComponent implements OnDestroy, OnInit {
 
     private async logOut(expired: boolean) {
         await this.eventService.uploadEvents();
-        const userId = await this.userService.getUserId();
-
+        const userId = await this.stateService.getUserId();
         await Promise.all([
             this.eventService.clearEvents(),
             this.syncService.setLastSync(new Date(0)),
             this.tokenService.clearToken(),
             this.cryptoService.clearKeys(),
-            this.userService.clear(),
             this.settingsService.clear(userId),
             this.cipherService.clear(userId),
             this.folderService.clear(userId),
             this.collectionService.clear(userId),
             this.policyService.clear(userId),
             this.passwordGenerationService.clear(),
-            this.stateService.purge(),
```

Comment:
It happens a little below this now on line 227. State needs to be the last thing that gets cleaned since it cleans up everything. I'm not totally sure we even still need all these `clear` calls, since stateService.clear() should remove all non-settings items from memory and storage, but I left them here not to create any more unnecessary risk.